apps/web/src/pages/EmailPage/EmailPage.tsx

```diff
@@ -11,38 +15,87 @@ type Props = {
 };
 
 export const EmailPage = ({ emails, on, close }: Props) => {
-  const { mutate: makeEmailsMutation } = useMakeEmails();
-
+  const { mutateAsync: makeEmailsMutation } = useMakeEmails();
+  const [subject, setSubject] = useState<string>('');
+  const [emailPreviewOpen, setEmailPreviewOpen] = useState<boolean>(false);
+  const [emailList, setEmailList] = useState<string[]>();
   const sendEmails = async (text: string) => {
-    makeEmailsMutation({ emails, text });
-    alert('emails sent, check logs	');
+    const createdEmails = await makeEmailsMutation({ emails, text });
+    console.log(createdEmails);
+    setEmailList(createdEmails);
+    setEmailPreviewOpen(true);
   };
 
   return (
-    <Modal open={on}>
+    <Modal
+      open={on}
+      sx={{
+        overflow: 'scroll',
+      }}
+    >
       <Box
         display="flex"
         flexDirection="column"
         alignItems="center"
         justifyContent="space-around"
         padding={'1rem'}
       >
-        Emails
-        <EmailBox sendEmail={sendEmails} />
-        <EmailList emails={emails} />
         <Button
           sx={{
-            width: '200px',
-            backgroundColor: 'red',
+            width: '50px',
+            height: '50px',
+            alignSelf: 'flex-end',
+            backgroundColor: 'transparent',
             color: 'white',
-            marginTop: '1rem',
-            transition: 'background-color 0.3s ease, color 0.3s ease',
-            '&:hover': { backgroundColor: 'gray', color: 'red' },
+            marginTop: '0.3rem',
+            transition: 'color 0.3s ease',
+            '&:hover': {
+              backgroundColor: 'transparent',
+              color: 'red',
+              img: { filter: 'invert(1)' },
+            },
+            img: {
+              filter: 'invert(0)',
+              transition: 'filter 0.3s ease',
+              height: '100%',
+              width: '100%',
+            },
           }}
           onClick={close}
         >
-          Close
+          <img src={xSymbol} alt="close" />
         </Button>
+        Emails
+        <Input
+          value={subject}
+          onChange={(e) => setSubject(e.target.value)}
+          inputMode="text"
+          sx={{
+            width: '66vw',
+            backgroundColor: '#D9D9D9',
+            marginBottom: '1rem',
+            display: 'flex',
+            flexDirection: 'column',
+            alignItems: 'flex-start',
+            input: {
+              padding: '1rem',
+              fontSize: '1.5rem',
+              fontWeight: 'bold',
+              width: '100%',
+            },
+          }}
+          placeholder="Subject"
+        />
+        <EmailBox sendEmail={sendEmails} />
+        <EmailPreviewList
+          subject={subject}
+          emails={emailList as string[]}
```

Comment:
maknit ovaj cast jer s razlogom baca error da je `emailList` nekad undefined i to treba riješiti dodavanjem defaultne vrijednosti u useState

Reply:
✅